### PR TITLE
break out the logic to a script and manual workflow

### LINKED ISSUE #1899: [FR] Convert backport logic to sh script
https://github.com/elastic/detection-rules/blob/bcec8a447992a8e49b421858ef8bf109f79006cf/.github/workflows/backport.yml#L110-L148

Then call it from the workflow as a script, passing the args.

This would allow us to manually call it when needed for simplicity

### PR DESCRIPTION
**Pull request by Mikaayenson**
## Issues

Resolves #1899 

## Summary
Break out the logic to a script and manual workflow for easier manual backporting.

DR Backporting Process

Example of what the workflow looks like.
Screen Shot 2022-04-21 at 12 15 02 PM

### CODE REVIEW

**Comment by Mikaayenson** [comment]
Once I test this E2E, I can update the backport.yml to also use the commit-and-push.sh script.

**Review by terrancedejesus — APPROVED** [review]
LGTM.

**Comment by Mikaayenson** [comment]
@brokensound77 do you want to also update `backport.yml` to use the new script like the `manual-backport.yml` or leave it as-is?

**Comment by Mikaayenson** [comment]
@brokensound77 do you want to also update `backport.yml` to use the new script like the `manual-backport.yml` or leave it as-is?

**Review by w0rk3r — APPROVED** [review]
LGTM

**Comment by Mikaayenson** [comment]
Screen Shot 2022-06-15 at 4 38 13 PM
Screen Shot 2022-06-15 at 4 37 49 PM

File: detection_rules/devtools.py [hunk 1]
```diff
@@ -195,11 +195,14 @@ def read(self, git_tree="HEAD") -> bytes:
...
 @click.option("--dry-run", is_flag=True, help="List the changes that would be made")
-def prune_staging_area(target_stack_version: str, dry_run: bool):
+@click.option("--exception-list", help="List of files to skip staging", default="")
+def prune_staging_area(target_stack_version: str, dry_run: bool, exception_list: list):
     """Prune the git staging area to remove changes to incompatible rules."""
     exceptions = {
         "detection_rules/etc/packages.yml",
     }
+    for exception in exception_list.split(","):
+        exceptions.add(exception)
```
Comment (brokensound77) [inline detection_rules/devtools.py:205]:
```suggestion
    exceptions.update(exception_list.split(","))
```

**Review by terrancedejesus — APPROVED** [review]
LGTM.

File: .github/workflows/manual-backport.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+jobs:
+  commit:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Validate the source branch
+        uses: actions/github-script@v3
+        with:
+          script: |
+            if ('refs/heads/main' !== '${{github.event.ref}}') {
```
Comment (brokensound77) [inline .github/workflows/manual-backport.yml:24]: Do we really want to limit this to only backporting from main?

File: .github/workflows/backport.yml [hunk 3]
```diff
@@ -110,46 +110,10 @@ jobs:
         env:
           COMMIT_MSG_FILE: "../commit-message.txt"
           UNSTAGED_LIST_FILE: "../unstaged-rules.txt"
+          TARGET_BRANCH: "${{github.event.inputs.target_branch}}"
+          COMMIT_SHA: "${{github.event.inputs.commit_sha}}"
```
Comment (brokensound77) [inline .github/workflows/backport.yml:114]: Why do these need to be passed for the existing backport?

File: .github/workflows/manual-backport.yml (hunk 2 above)
Comment (Mikaayenson) [reply .github/workflows/manual-backport.yml]: Great question. We shouldn't do this anymore. I removed this main check.

File: detection_rules/etc/commit-and-push.sh [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+git checkout ${TARGET_BRANCH}
+
+echo "Applying new changes"
+(git stash apply --quiet) || true
+
+echo "Selecting incoming changes to be committed"
+git checkout --theirs . --quiet
+
+echo "Track new changes"
+git add -A
```
Comment (Mikaayenson) [inline detection_rules/etc/commit-and-push.sh:23]: This is to resolve merge conflicts.

**Comment by brokensound77** [comment]
>[reset backport.yml](https://github.com/elastic/detection-rules/pull/1908/commits/16095f514986e9c9355ca9a159b76e94bf3a51b7)

I thought based on our last discussion that you were going to leave backport.yml as is and keep the manual workflow completely separate for now for mergeability. It looks like this did the opposite?

**Comment by Mikaayenson** [comment]
## Update

Based on the last discussion, we are ready to merge this in with low regret as the existing back porting workflows will not change. This PR adds a manual backport option which will be used if the backporting process fails unexpectedly in the future. 

E.g. GitHub operations go down during the backport process. 

Instead of manually fixing the backport branches with command-line git commands, we can use this workflow to automate the process.

**Comment by Mikaayenson** [comment]
> > [reset backport.yml](https://github.com/elastic/detection-rules/pull/1908/commits/16095f514986e9c9355ca9a159b76e94bf3a51b7)
> 
> I thought based on our last discussion that you were going to leave backport.yml as is and keep the manual workflow completely separate for now for mergeability. It looks like this did the opposite?

Yup that's exactly right, I removed the changes from this branch. Now there are only three files in the Files Changed..

Screen Shot 2022-09-16 at 1 31 22 PM

**Comment by Mikaayenson** [comment]
## Update

Based on the last discussion, we are ready to merge this in with low regret as the existing back porting workflows will not change. This PR adds a manual backport option which will be used if the backporting process fails unexpectedly in the future. 

E.g. GitHub operations go down during the backport process. 

Instead of manually fixing the backport branches with command-line git commands, we can use this workflow to automate the process. Once we test more, we can then consider refactoring the original backport CI workflow to use the commit-and-push.sh script.

**Pull request merged**